File: app/src/main/java/social/entourage/android/user/UserComponent.java
package social.entourage.android.user;

import dagger.Component;
import social.entourage.android.ActivityScope;
import social.entourage.android.EntourageComponent;

/**
 * Component linked to UserActivity lifecycle
 * Provide a UserPresenter
 * @see UserActivity
 * @see UserPresenter
 */
@ActivityScope
@Component(
        dependencies = EntourageComponent.class,
        modules = UserModule.class
)
@SuppressWarnings("unused")
public interface UserComponent {
    void inject(UserActivity activity);

    UserPresenter getUserPresenter();
}

File: app/src/main/java/social/entourage/android/user/UserComponent.kt
package social.entourage.android.user

import dagger.Component
import social.entourage.android.base.ActivityScope
import social.entourage.android.EntourageComponent

/**
 * Component linked to UserFragment lifecycle
 * Provides a UserPresenter
 * @see UserFragment
 *
 * @see UserPresenter
 */
@ActivityScope
@Component(dependencies = [EntourageComponent::class], modules = [UserModule::class])
interface UserComponent {
    fun inject(fragment: UserFragment?)
    val userPresenter: UserPresenter?
}